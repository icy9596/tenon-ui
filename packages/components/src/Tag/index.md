---
title: Tag
demo:
  cols: 2
nav:
  title: 组件
group:
  title: 通用
---

# Tag 标签

进行标记和分类的小标签。

## 代码演示

<code src="./demo/base.tsx">基本</code>
