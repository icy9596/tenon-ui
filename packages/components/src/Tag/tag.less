.tenon-tag {
  background-color: #f0f0f0;
  border: 1px solid #ccc;
  padding: 2px 10px;
}
