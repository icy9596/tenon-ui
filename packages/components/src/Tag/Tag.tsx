import './tag.less';

export interface TagProps {
  children: string;
}

const Tag = ({ children }: TagProps): JSX.Element => {
  return <span className="tenon-tag">{children}</span>;
};

export default Tag;
