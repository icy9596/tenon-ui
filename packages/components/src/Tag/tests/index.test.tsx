import { render, screen } from '@testing-library/react';

import Tag from '../Tag';

const tagText = 'test';

describe('Tag', () => {
  it('render', () => {
    const { container } = render(<Tag>{tagText}</Tag>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('mount', () => {
    render(<Tag>{tagText}</Tag>);
    expect(screen.getByText(tagText)).toBeInTheDocument();
  });
});
