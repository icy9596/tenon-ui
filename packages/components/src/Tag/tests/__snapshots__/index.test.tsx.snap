// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag render 1`] = `
<span
  class="tenon-tag"
>
  test
</span>
`;
